In [1]:
# import libraries

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

plt.style.use('ggplot')
pd.set_option('display.max_columns', 200) 

In [2]:
df = pd.read_csv('acs2017_county_data.csv', sep=',').copy()
df

CountyId        State               County  TotalPop    Men   Women  \
0         1001      Alabama       Autauga County     55036  26899   28137   
1         1003      Alabama       Baldwin County    203360  99527  103833   
2         1005      Alabama       Barbour County     26201  13976   12225   
3         1007      Alabama          Bibb County     22580  12251   10329   
4         1009      Alabama        Blount County     57667  28490   29177   
...        ...          ...                  ...       ...    ...     ...   
3215     72145  Puerto Rico  Vega Baja Municipio     54754  26269   28485   
3216     72147  Puerto Rico    Vieques Municipio      8931   4351    4580   
3217     72149  Puerto Rico   Villalba Municipio     23659  11510   12149   
3218     72151  Puerto Rico    Yabucoa Municipio     35025  16984   18041   
3219     72153  Puerto Rico      Yauco Municipio     37585  18052   19533   

      Hispanic  White  Black  Native  Asian  Pacific  VotingAgeCitizen  \
0          2.7   75.4   18.9     0.3    0.9      0.0             41016   
1          4.4   83.1    9.5     0.8    0.7      0.0            155376   
2          4.2   45.7   47.8     0.2    0.6      0.0             20269   
3          2.4   74.6   22.0     0.4    0.0      0.0             17662   
4          9.0   87.4    1.5     0.3    0.1      0.0             42513   
...        ...    ...    ...     ...    ...      ...               ...   
3215      96.7    3.1    0.1     0.0    0.0      0.0             42838   
3216      95.7    4.0    0.0     0.0    0.0      0.0              7045   
3217      99.7    0.2    0.1     0.0    0.0      0.0             18053   
3218      99.9    0.1    0.0     0.0    0.0      0.0             27523   
3219      99.8    0.2    0.0     0.0    0.0      0.0             29763   

      Income  IncomeErr  IncomePerCap  IncomePerCapErr  Poverty  ChildPoverty  \
0      55317       2838         27824             2024     13.7          20.1   
1      52562       1348         29364              735     11.8          16.1   
2      33368       2551         17561              798     27.2          44.9   
3      43404       3431         20911             1889     15.2          26.6   
4      47412       2630         22021              850     15.6          25.4   
...      ...        ...           ...              ...      ...           ...   
3215   18900       1219         10197              576     43.8          49.4   
3216   16261       2414         11136             1459     36.8          68.2   
3217   19893       1935         10449             1619     50.0          67.9   
3218   15586       1467          8672              702     52.4          62.1   
3219   14451       1371          8124              445     50.4          58.2   

      Professional  Service  Office  Construction  Production  Drive  Carpool  \
0             35.3     18.0    23.2           8.1        15.4   86.0      9.6   
1             35.7     18.2    25.6           9.7        10.8   84.7      7.6   
2             25.0     16.8    22.6          11.5        24.1   83.4     11.1   
3             24.4     17.6    19.7          15.9        22.4   86.4      9.5   
4             28.5     12.9    23.3          15.8        19.5   86.8     10.2   
...            ...      ...     ...           ...         ...    ...      ...   
3215          28.6     20.2    25.9          11.1        14.2   92.0      4.2   
3216          20.9     38.4    16.4          16.9         7.3   76.3     16.9   
3217          22.5     21.2    22.7          14.1        19.5   83.1     11.8   
3218          27.7     26.0    20.7           9.5        16.0   87.6      9.2   
3219          30.4     20.2    25.6          12.6        11.3   82.8      8.2   

      Transit  Walk  OtherTransp  WorkAtHome  MeanCommute  Employed  \
0         0.1   0.6          1.3         2.5         25.8     24112   
1         0.1   0.8          1.1         5.6         27.0     89527   
2         0.3   2.2          1.7 

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3220 entries, 0 to 3219
Data columns (total 37 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   CountyId          3220 non-null   int64  
 1   State             3220 non-null   object 
 2   County            3220 non-null   object 
 3   TotalPop          3220 non-null   int64  
 4   Men               3220 non-null   int64  
 5   Women             3220 non-null   int64  
 6   Hispanic          3220 non-null   float64
 7   White             3220 non-null   float64
 8   Black             3220 non-null   float64
 9   Native            3220 non-null   float64
 10  Asian             3220 non-null   float64
 11  Pacific           3220 non-null   float64
 12  VotingAgeCitizen  3220 non-null   int64  
 13  Income            3220 non-null   int64  
 14  IncomeErr         3220 non-null   int64  
 15  IncomePerCap      3220 non-null   int64  
 16  IncomePerCapErr   3220 non-null   int64  


In [4]:
df.columns

Index(['CountyId', 'State', 'County', 'TotalPop', 'Men', 'Women', 'Hispanic',
       'White', 'Black', 'Native', 'Asian', 'Pacific', 'VotingAgeCitizen',
       'Income', 'IncomeErr', 'IncomePerCap', 'IncomePerCapErr', 'Poverty',
       'ChildPoverty', 'Professional', 'Service', 'Office', 'Construction',
       'Production', 'Drive', 'Carpool', 'Transit', 'Walk', 'OtherTransp',
       'WorkAtHome', 'MeanCommute', 'Employed', 'PrivateWork', 'PublicWork',
       'SelfEmployed', 'FamilyWork', 'Unemployment'],
      dtype='object')

In [5]:
# drop the columns which are not needed 

df.drop(columns=['CountyId', 'IncomeErr', 'IncomePerCapErr',  
                'Hispanic', 'White', 'Black', 'Native', 'Asian', 'Pacific', 
                'Professional', 'Service', 'Office', 'Construction','Production',
                'Drive', 'Carpool', 'Transit', 'Walk', 'OtherTransp',
                'WorkAtHome', 'MeanCommute', 'PrivateWork', 'PublicWork', 'SelfEmployed', 'FamilyWork', ], inplace=True)
df.head()

State          County  TotalPop    Men   Women  VotingAgeCitizen  Income  \
0  Alabama  Autauga County     55036  26899   28137             41016   55317   
1  Alabama  Baldwin County    203360  99527  103833            155376   52562   
2  Alabama  Barbour County     26201  13976   12225             20269   33368   
3  Alabama     Bibb County     22580  12251   10329             17662   43404   
4  Alabama   Blount County     57667  28490   29177             42513   47412   

   IncomePerCap  Poverty  ChildPoverty  Employed  Unemployment  
0         27824     13.7          20.1     24112           5.2  
1         29364     11.8          16.1     89527           5.5  
2         17561     27.2          44.9      8878          12.4  
3         20911     15.2          26.6      8171           8.2  
4         22021     15.6          25.4     21380           4.9

### Review features and values

In [6]:
df['State'].unique()

array(['Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California',
       'Colorado', 'Connecticut', 'Delaware', 'District of Columbia',
       'Florida', 'Georgia', 'Hawaii', 'Idaho', 'Illinois', 'Indiana',
       'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine', 'Maryland',
       'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi',
       'Missouri', 'Montana', 'Nebraska', 'Nevada', 'New Hampshire',
       'New Jersey', 'New Mexico', 'New York', 'North Carolina',
       'North Dakota', 'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania',
       'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee',
       'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington',
       'West Virginia', 'Wisconsin', 'Wyoming', 'Puerto Rico'],
      dtype=object)

In [7]:
df['State'].nunique()

52

* I am keeping District of Columbia as if it is a state to represent it, but removing Puerto Rico to focus on USA states

In [8]:
df = df[df['State'] != 'Puerto Rico'].reset_index(drop=True)

df.head()

State          County  TotalPop    Men   Women  VotingAgeCitizen  Income  \
0  Alabama  Autauga County     55036  26899   28137             41016   55317   
1  Alabama  Baldwin County    203360  99527  103833            155376   52562   
2  Alabama  Barbour County     26201  13976   12225             20269   33368   
3  Alabama     Bibb County     22580  12251   10329             17662   43404   
4  Alabama   Blount County     57667  28490   29177             42513   47412   

   IncomePerCap  Poverty  ChildPoverty  Employed  Unemployment  
0         27824     13.7          20.1     24112           5.2  
1         29364     11.8          16.1     89527           5.5  
2         17561     27.2          44.9      8878          12.4  
3         20911     15.2          26.6      8171           8.2  
4         22021     15.6          25.4     21380           4.9

In [9]:
df['County'].nunique()

1877

* there are over 3000 counties in USA but here we have less

In [10]:
df.rename(columns={'TotalPop': 'Population'}, inplace=True)

In [11]:
# review population men and women values
df['check_population'] = df['Men'] + df['Women'] == df['Population']

df[~df['check_population']]

Empty DataFrame
Columns: [State, County, Population, Men, Women, VotingAgeCitizen, Income, IncomePerCap, Poverty, ChildPoverty, Employed, Unemployment, check_population]
Index: []

In [12]:
df.drop(columns=[ 'check_population' ], inplace=True)

In [13]:
df.rename(columns={
    'IncomePerCap': 'IncomePerCapita',
    'Men': 'MenPopulation',
    'Women': 'WomenPopulation',
    'Poverty': 'PovertyRate',
    'ChildPoverty': 'ChildPovertyRate',
    'Employed': 'EmployedPopulation',
    'Unemployment': 'UnemploymentRate'
}, inplace=True)

In [14]:
df.head().T

0               1               2  \
State                      Alabama         Alabama         Alabama   
County              Autauga County  Baldwin County  Barbour County   
Population                   55036          203360           26201   
MenPopulation                26899           99527           13976   
WomenPopulation              28137          103833           12225   
VotingAgeCitizen             41016          155376           20269   
Income                       55317           52562           33368   
IncomePerCapita              27824           29364           17561   
PovertyRate                   13.7            11.8            27.2   
ChildPovertyRate              20.1            16.1            44.9   
EmployedPopulation           24112           89527            8878   
UnemploymentRate               5.2             5.5            12.4   

                              3              4  
State                   Alabama        Alabama  
County              Bibb County  Blount County  
Population                22580          57667  
MenPopulation             12251          28490  
WomenPopulation           10329          29177  
VotingAgeCitizen          17662          42513  
Income                    43404          47412  
IncomePerCapita           20911          22021  
PovertyRate                15.2           15.6  
ChildPovertyRate           26.6           25.4  
EmployedPopulation         8171          21380  
UnemploymentRate            8.2            4.9

In [15]:
# check if VotingAgeCitizen is smaller than EmployedPopulation in any record, which shouldnt be

df[df['VotingAgeCitizen'] < df['EmployedPopulation']]

State                      County  Population  MenPopulation  \
67  Alaska      Aleutians East Borough        3338           2049   
68  Alaska  Aleutians West Census Area        5784           3539   

    WomenPopulation  VotingAgeCitizen  Income  IncomePerCapita  PovertyRate  \
67             1289              2057   66607            31254         16.7   
68             2245              3565   85192            35998          7.5   

    ChildPovertyRate  EmployedPopulation  UnemploymentRate  
67              20.0                2405               2.4  
68               9.9                3961               1.9

In [16]:
# correct the EmployedPopulation based on VotingAgeCitizen and UnemploymentRate

df.loc[(df['State'] == 'Alaska') & (df['County'] == 'Aleutians East Borough'), 'EmployedPopulation'] = round(df['VotingAgeCitizen'] * (1 - (df['UnemploymentRate'] / 100)))
df.loc[(df['State'] == 'Alaska') & (df['County'] == 'Aleutians West Census Area'), 'EmployedPopulation'] = round(df['VotingAgeCitizen'] * (1 - (df['UnemploymentRate'] / 100)))

In [17]:
df[df['County'] == 'Aleutians East Borough']

State                  County  Population  MenPopulation  \
67  Alaska  Aleutians East Borough        3338           2049   

    WomenPopulation  VotingAgeCitizen  Income  IncomePerCapita  PovertyRate  \
67             1289              2057   66607            31254         16.7   

    ChildPovertyRate  EmployedPopulation  UnemploymentRate  
67              20.0                2008               2.4

In [18]:
df[df['County'] == 'Aleutians West Census Area']


State                      County  Population  MenPopulation  \
68  Alaska  Aleutians West Census Area        5784           3539   

    WomenPopulation  VotingAgeCitizen  Income  IncomePerCapita  PovertyRate  \
68             2245              3565   85192            35998          7.5   

    ChildPovertyRate  EmployedPopulation  UnemploymentRate  
68               9.9                3497               1.9

In [19]:
df.describe(include='all').T

count unique                top freq           mean  \
State                 3142     51              Texas  254            NaN   
County                3142   1877  Washington County   30            NaN   
Population          3142.0    NaN                NaN  NaN  102165.629217   
MenPopulation       3142.0    NaN                NaN  NaN   50292.410248   
WomenPopulation     3142.0    NaN                NaN  NaN   51873.218969   
VotingAgeCitizen    3142.0    NaN                NaN  NaN   72223.490452   
Income              3142.0    NaN                NaN  NaN   49754.163908   
IncomePerCapita     3142.0    NaN                NaN  NaN   26039.637174   
PovertyRate         3142.0    NaN                NaN  NaN      15.987651   
ChildPovertyRate    3141.0    NaN                NaN  NaN      22.106749   
EmployedPopulation  3142.0    NaN                NaN  NaN   47930.714195   
UnemploymentRate    3142.0    NaN                NaN  NaN       6.364004   

                              std      min       25%      50%       75%  \
State                         NaN      NaN       NaN      NaN       NaN   
County                        NaN      NaN       NaN      NaN       NaN   
Population          328292.195145     74.0  10945.25  25691.5   67445.0   
MenPopulation       161184.958332     39.0    5514.5  12798.0   33481.0   
WomenPopulation      167145.28419     35.0    5460.0  12885.0  34107.75   
VotingAgeCitizen    213311.374873     59.0   8279.25  19479.5  51223.75   
Income               13154.424871  19264.0   41123.0  48065.5   55763.5   
IncomePerCapita       6270.035387   9334.0   21809.5  25272.5   29125.5   
PovertyRate              6.558057      2.4      11.3     15.2      19.4   
ChildPovertyRate        10.300409      0.0      14.8     21.2      27.8   
EmployedPopulation  157622.109759     39.0    4550.5  10695.0  29514.75   
UnemploymentRate         3.063165      0.0       4.4      6.1       7.8   

                           max  
State                      NaN  
County                     NaN  
Population          10105722.0  
MenPopulation        4979641.0  
WomenPopulation      5126081.0  
VotingAgeCitizen     6218279.0  
Income                129588.0  
IncomePerCapita        69529.0  
PovertyRate               52.0  
ChildPovertyRate          76.5  
EmployedPopulation   4805817.0  
UnemploymentRate          28.8

In [20]:
df[df['UnemploymentRate'] == 0]

State            County  Population  MenPopulation  \
548       Hawaii    Kalawao County          86             41   
1614     Montana   Garfield County        1086            560   
1649     Montana   Treasure County         790            377   
1712    Nebraska  McPherson County         432            221   
1736    Nebraska      Sioux County        1256            647   
1752      Nevada     Eureka County        1728            849   
1796  New Mexico     Catron County        3547           1861   
2609       Texas  Glasscock County        1420            792   
2653       Texas     Kenedy County         564            280   
2673       Texas     Loving County          74             39   
3132     Wyoming   Niobrara County        2479           1098   

      WomenPopulation  VotingAgeCitizen  Income  IncomePerCapita  PovertyRate  \
548                45                84   61750            46024         12.7   
1614              526               861   47188            29896          5.1   
1649              413               634   42454            22880         17.0   
1712              211               335   60714            27241         16.0   
1736              609               998   45375            26852         12.4   
1752              879              1306   67159            35606         10.0   
1796             1686              3023   42047            22487         21.5   
2609              628               855   64886            32885         11.6   
2653              284               296   24800            13705         24.8   
2673               35                59   80938            35530         17.1   
3132             1381              1999   36793            23370         14.9   

      ChildPovertyRate  EmployedPopulation  UnemploymentRate  
548                NaN                  63               0.0  
1614               1.8                 576               0.0  
1649              25.6                 388               0.0  
1712              21.6                 225               0.0  
1736              18.7                 660               0.0  
1752               0.0                 904               0.0  
1796              33.1                 919               0.0  
2609               6.1                 669               0.0  
2653              29.9                 183               0.0  
2673               0.0                  39               0.0  
3132              13.5                1067               0.0

In [21]:
# correct EmployedPopulation where UnemploymentRate is 0

df.loc[df['UnemploymentRate'] == 0, 'UnemploymentRate'] = round(((df['VotingAgeCitizen'] - df['EmployedPopulation']) / df['VotingAgeCitizen']) * 100, 1)

In [22]:
# check the extreme values for UnemploymentRate
df[df['UnemploymentRate'] > 50]

State         County  Population  MenPopulation  WomenPopulation  \
1796  New Mexico  Catron County        3547           1861             1686   

      VotingAgeCitizen  Income  IncomePerCapita  PovertyRate  \
1796              3023   42047            22487         21.5   

      ChildPovertyRate  EmployedPopulation  UnemploymentRate  
1796              33.1                 919              69.6

* 69,6 % UnemploymentRate for 2017 seems unrealistic and likely due to a data entry issue. 
* however, considering the data source specifies 919 employed individuals among 3,023 VotingAgeCitizens, it appears intentional for practice purposes, so I am keeping it.

In [23]:
# calculate UnemployedPopulation
df['UnemployedPopulation'] = (df['VotingAgeCitizen'] * df['UnemploymentRate'] / 100).round().astype(int)

# calculate EmploymentRate
df['EmploymentRate'] = (df['EmployedPopulation'] / df['VotingAgeCitizen'] * 100).round(1)

In [24]:
df.isna().sum()

State                   0
County                  0
Population              0
MenPopulation           0
WomenPopulation         0
VotingAgeCitizen        0
Income                  0
IncomePerCapita         0
PovertyRate             0
ChildPovertyRate        1
EmployedPopulation      0
UnemploymentRate        0
UnemployedPopulation    0
EmploymentRate          0
dtype: int64

In [25]:
df[df['ChildPovertyRate'].isna()]

State          County  Population  MenPopulation  WomenPopulation  \
548  Hawaii  Kalawao County          86             41               45   

     VotingAgeCitizen  Income  IncomePerCapita  PovertyRate  ChildPovertyRate  \
548                84   61750            46024         12.7               NaN   

     EmployedPopulation  UnemploymentRate  UnemployedPopulation  \
548                  63              25.0                    21   

     EmploymentRate  
548            75.0

In [26]:
df_filtered = df[df['ChildPovertyRate'] == 0]
df_filtered.head(20).T

300            1752          2398  \
State                        Colorado         Nevada  South Dakota   
County                San Juan County  Eureka County  Jones County   
Population                        569           1728           740   
MenPopulation                     301            849           375   
WomenPopulation                   268            879           365   
VotingAgeCitizen                  475           1306           617   
Income                          46397          67159         39432   
IncomePerCapita                 28889          35606         26526   
PovertyRate                       3.3           10.0          10.8   
ChildPovertyRate                  0.0            0.0           0.0   
EmployedPopulation                305            904           430   
UnemploymentRate                  9.8           30.8           2.9   
UnemployedPopulation               47            402            18   
EmploymentRate                   64.2           69.2          69.7   

                               2539               2644           2673  \
State                         Texas              Texas          Texas   
County                Borden County  Jeff Davis County  Loving County   
Population                      602               2236             74   
MenPopulation                   309               1070             39   
WomenPopulation                 293               1166             35   
VotingAgeCitizen                435               1542             59   
Income                        77708              46534          80938   
IncomePerCapita               38923              25167          35530   
PovertyRate                     2.8                6.7           17.1   
ChildPovertyRate                0.0                0.0            0.0   
EmployedPopulation              255                911             39   
UnemploymentRate                1.5                5.1           33.9   
UnemployedPopulation              7                 79             20   
EmploymentRate                 58.6               59.1           66.1   

                                 2678            2781  
State                           Texas            Utah  
County                McMullen County  Daggett County  
Population                        600             702  
MenPopulation                     321             425  
WomenPopulation                   279             277  
VotingAgeCitizen                  406             539  
Income                          71389           85000  
IncomePerCapita                 33472           28533  
PovertyRate                       7.0             4.0  
ChildPovertyRate                  0.0             0.0  
EmployedPopulation                202             293  
UnemploymentRate                 26.3             1.0  
UnemployedPopulation              107               5  
EmploymentRate                   49.8            54.4

In [27]:
filtered_df = df[df['UnemployedPopulation'] < 10]
filtered_df.head(10).T

261            1656           1658  \
State                       Colorado       Nebraska       Nebraska   
County                Dolores County  Arthur County  Blaine County   
Population                      1736            421            502   
MenPopulation                    946            204            256   
WomenPopulation                  790            217            246   
VotingAgeCitizen                1362            289            396   
Income                         40352          41250          52000   
IncomePerCapita                23190          21799          28503   
PovertyRate                     14.1           10.9            9.8   
ChildPovertyRate                19.7            8.3           15.1   
EmployedPopulation               797            177            282   
UnemploymentRate                 0.4            0.6            1.1   
UnemployedPopulation               5              2              4   
EmploymentRate                  58.5           61.2           71.2   

                               1699              1705          1710  \
State                      Nebraska          Nebraska      Nebraska   
County                Hooker County  Keya Paha County  Logan County   
Population                      684               720           919   
MenPopulation                   315               368           471   
WomenPopulation                 369               352           448   
VotingAgeCitizen                578               575           644   
Income                        41607             48864         50278   
IncomePerCapita               27714             28981         24624   
PovertyRate                    17.4               7.7           9.7   
ChildPovertyRate               24.3              13.3          12.2   
EmployedPopulation              304               428           450   
UnemploymentRate                1.0               1.2           1.1   
UnemployedPopulation              6                 7             7   
EmploymentRate                 52.6              74.4          69.9   

                             1711            1745          2033          2385  
State                    Nebraska        Nebraska  North Dakota  South Dakota  
County                Loup County  Wheeler County  Slope County  Faulk County  
Population                    537             846           674          2327  
MenPopulation                 260             444           350          1196  
WomenPopulation               277             402           324          1131  
VotingAgeCitizen              433             686           528          1839  
Income                      58625           49706         61250         46750  
IncomePerCapita             29248           28257         35118         29714  
PovertyRate                   8.9             2.6           6.8          10.9  
ChildPovertyRate              9.6             3.2           2.8           0.9  
EmployedPopulation            295             494           307          1120  
UnemploymentRate              0.7             0.4           1.3           0.3  
UnemployedPopulation            3               3             7             6  
EmploymentRate               68.1            72.0          58.1          60.9

In [28]:
filtered_df = df[df['UnemploymentRate'] < 1]
filtered_df.head(10).T

261           985                 1646  \
State                       Colorado        Kansas             Montana   
County                Dolores County  Trego County  Sweet Grass County   
Population                      1736          2899                3646   
MenPopulation                    946          1465                1763   
WomenPopulation                  790          1434                1883   
VotingAgeCitizen                1362          2317                2801   
Income                         40352         55197               49761   
IncomePerCapita                23190         33233               28065   
PovertyRate                     14.1           4.8                10.4   
ChildPovertyRate                19.7           2.7                15.9   
EmployedPopulation               797          1747                1621   
UnemploymentRate                 0.4           0.8                 0.9   
UnemployedPopulation               5            19                  25   
EmploymentRate                  58.5          75.4                57.9   

                               1656          1662         1711           1722  \
State                      Nebraska      Nebraska     Nebraska       Nebraska   
County                Arthur County  Brown County  Loup County  Phelps County   
Population                      421          3058          537           9163   
MenPopulation                   204          1468          260           4603   
WomenPopulation                 217          1590          277           4560   
VotingAgeCitizen                289          2401          433           6795   
Income                        41250         42773        58625          56408   
IncomePerCapita               21799         29664        29248          30166   
PovertyRate                    10.9          16.0          8.9            7.1   
ChildPovertyRate                8.3          22.3          9.6            6.9   
EmployedPopulation              177          1584          295           4780   
UnemploymentRate                0.6           0.6          0.7            0.7   
UnemployedPopulation              2            14            3             48   
EmploymentRate                 61.2          66.0         68.1           70.3   

                                1745                  2006            2012  
State                       Nebraska          North Dakota    North Dakota  
County                Wheeler County  Golden Valley County  LaMoure County  
Population                       846                  1892            4106  
MenPopulation                    444                   930            2129  
WomenPopulation                  402                   962            1977  
VotingAgeCitizen                 686                  1416            3175  
Income                         49706                 44018           57463  
IncomePerCapita                28257                 28840           36653  
PovertyRate                      2.6                   9.4            10.7  
ChildPovertyRate                 3.2                  14.8            15.3  
EmployedPopulation               494                  1012            2019  
UnemploymentRate                 0.4                   0.9             0.8  
UnemployedPopulation               3                    13              25  
EmploymentRate                  72.0                  71.5            63.6

In [29]:
filtered_df = df[df['UnemploymentRate'] > 25]
filtered_df.head(50).T

81               1614             1649  \
State                               Alaska          Montana          Montana   
County                Kusilvak Census Area  Garfield County  Treasure County   
Population                            8129             1086              790   
MenPopulation                         4342              560              377   
WomenPopulation                       3787              526              413   
VotingAgeCitizen                      4813              861              634   
Income                               36468            47188            42454   
IncomePerCapita                      11693            29896            22880   
PovertyRate                           39.1              5.1             17.0   
ChildPovertyRate                      45.8              1.8             25.6   
EmployedPopulation                    2125              576              388   
UnemploymentRate                      28.8             33.1             38.8   
UnemployedPopulation                  1386              285              246   
EmploymentRate                        44.2             66.9             61.2   

                                  1712          1736           1752  \
State                         Nebraska      Nebraska         Nevada   
County                McPherson County  Sioux County  Eureka County   
Population                         432          1256           1728   
MenPopulation                      221           647            849   
WomenPopulation                    211           609            879   
VotingAgeCitizen                   335           998           1306   
Income                           60714         45375          67159   
IncomePerCapita                  27241         26852          35606   
PovertyRate                       16.0          12.4           10.0   
ChildPovertyRate                  21.6          18.7            0.0   
EmployedPopulation                 225           660            904   
UnemploymentRate                  32.8          33.9           30.8   
UnemployedPopulation               110           338            402   
EmploymentRate                    67.2          66.1           69.2   

                               1796           2376          2381  \
State                    New Mexico   South Dakota  South Dakota   
County                Catron County  Corson County  Dewey County   
Population                     3547           4177          5709   
MenPopulation                  1861           2125          2823   
WomenPopulation                1686           2052          2886   
VotingAgeCitizen               3023           2692          3669   
Income                        42047          32260         41190   
IncomePerCapita               22487          15160         17267   
PovertyRate                    21.5           47.9          31.4   
ChildPovertyRate               33.1           61.7          36.4   
EmployedPopulation              919           1260          1950   
UnemploymentRate               69.6           28.7          25.9   
UnemployedPopulation           2104            773           950   
EmploymentRate                 30.4           46.8          53.1   

                                2427           2653           2673  \
State                   South Dakota          Texas          Texas   
County                Ziebach County  Kenedy County  Loving County   
Population                      2837            564             74   
MenPopulation                   1421            280             39   
WomenPopulation                 1416            284             35   
VotingAgeCitizen                1865            296             59   
Income                         35000          24800          80938   
IncomePerCapita                13461          13705          35530   
PovertyRate                     43.1           24.8           17.1   
ChildPovertyRate                55.6           29.

* again, despite many of these extreme values seeming unrealistic in real life, I am keeping them as they are, since they reflect the data source information

In [30]:
df.columns

Index(['State', 'County', 'Population', 'MenPopulation', 'WomenPopulation',
       'VotingAgeCitizen', 'Income', 'IncomePerCapita', 'PovertyRate',
       'ChildPovertyRate', 'EmployedPopulation', 'UnemploymentRate',
       'UnemployedPopulation', 'EmploymentRate'],
      dtype='object')

In [31]:
df = df[['State', 'County', 'Population', 'MenPopulation', 'WomenPopulation',
       'VotingAgeCitizen', 'Income', 'IncomePerCapita', 'PovertyRate',
       'ChildPovertyRate', 'EmployedPopulation', 'UnemployedPopulation',
       'EmploymentRate', 'UnemploymentRate']].reset_index(drop= True)

df

State             County  Population  MenPopulation  WomenPopulation  \
0     Alabama     Autauga County       55036          26899            28137   
1     Alabama     Baldwin County      203360          99527           103833   
2     Alabama     Barbour County       26201          13976            12225   
3     Alabama        Bibb County       22580          12251            10329   
4     Alabama      Blount County       57667          28490            29177   
...       ...                ...         ...            ...              ...   
3137  Wyoming  Sweetwater County       44527          22981            21546   
3138  Wyoming       Teton County       22923          12169            10754   
3139  Wyoming       Uinta County       20758          10593            10165   
3140  Wyoming    Washakie County        8253           4118             4135   
3141  Wyoming      Weston County        7117           3756             3361   

      VotingAgeCitizen  Income  IncomePerCapita  PovertyRate  \
0                41016   55317            27824         13.7   
1               155376   52562            29364         11.8   
2                20269   33368            17561         27.2   
3                17662   43404            20911         15.2   
4                42513   47412            22021         15.6   
...                ...     ...              ...          ...   
3137             31025   71083            31700         12.0   
3138             16692   80049            49200          6.8   
3139             14235   54672            27115         14.9   
3140              6125   51362            27345         12.8   
3141              5510   59605            30955         14.4   

      ChildPovertyRate  EmployedPopulation  UnemployedPopulation  \
0                 20.1               24112                  2133   
1                 16.1               89527                  8546   
2                 44.9                8878                  2513   
3                 26.6                8171                  1448   
4                 25.4               21380                  2083   
...                ...                 ...                   ...   
3137              15.7               22739                  1613   
3138               2.8               14492                   217   
3139              20.0                9528                   911   
3140              17.5                3833                   374   
3141              24.1                3407                   121   

      EmploymentRate  UnemploymentRate  
0               58.8               5.2  
1               57.6               5.5  
2               43.8              12.4  
3               46.3               8.2  
4               50.3               4.9  
...              ...               ...  
3137            73.3               5.2  
3138            86.8               1.3  
3139            66.9               6.4  
3140            62.6               6.1  
3141            61.8               2.2  

[3142 rows x 14 columns]

In [32]:
df.head().T

0               1               2  \
State                        Alabama         Alabama         Alabama   
County                Autauga County  Baldwin County  Barbour County   
Population                     55036          203360           26201   
MenPopulation                  26899           99527           13976   
WomenPopulation                28137          103833           12225   
VotingAgeCitizen               41016          155376           20269   
Income                         55317           52562           33368   
IncomePerCapita                27824           29364           17561   
PovertyRate                     13.7            11.8            27.2   
ChildPovertyRate                20.1            16.1            44.9   
EmployedPopulation             24112           89527            8878   
UnemployedPopulation            2133            8546            2513   
EmploymentRate                  58.8            57.6            43.8   
UnemploymentRate                 5.2             5.5            12.4   

                                3              4  
State                     Alabama        Alabama  
County                Bibb County  Blount County  
Population                  22580          57667  
MenPopulation               12251          28490  
WomenPopulation             10329          29177  
VotingAgeCitizen            17662          42513  
Income                      43404          47412  
IncomePerCapita             20911          22021  
PovertyRate                  15.2           15.6  
ChildPovertyRate             26.6           25.4  
EmployedPopulation           8171          21380  
UnemployedPopulation         1448           2083  
EmploymentRate               46.3           50.3  
UnemploymentRate              8.2            4.9

In [34]:
# save the to use to create a Tableau dashboard
df.to_csv('acs2017_county_clean.csv', index=False)

print("file saved")

file saved
